mypy/messages.py

```diff
@@ -1233,13 +1243,17 @@ def report_protocol_problems(self, subtype: Union[Instance, TupleType, TypedDict
                           .format(supertype.type.name(), name), context)
         self.print_more(conflict_flags, context, OFFSET, MAX_ITEMS)
 
-    def pretty_overload(self, tp: Overloaded, context: Context,
+    def pretty_overload(self, targets: List[CallableType], func: Overloaded, context: Context,
                         offset: int, max_items: int) -> None:
-        for item in tp.items()[:max_items]:
+        if not targets:
+            targets = func.items()
+        for item in targets[:max_items]:
             self.note('@overload', context, offset=2 * offset)
             self.note(self.pretty_callable(item), context, offset=2 * offset)
-        if len(tp.items()) > max_items:
-            self.note('<{} more overload(s) not shown>'.format(len(tp.items()) - max_items),
+        shown = min(max_items, len(targets))
+        max_available = len(func.items())
+        if shown < max_available:
+            self.note('<{} more overload(s) not shown>'.format(max_available - shown),
```

Comment:
Apart form the uglines around plurals, it seems there are two possible reasons why not all overloads are shown:

- because they don't match the number of arguments (or the keywords or whatever)
- because the sheer number exceeds `max_items`

I think it would be good to differentiate between these in the messages somehow.